statistics/update.go

```diff
@@ -133,3 +137,62 @@ func (h *Handle) dumpTableStatDeltaToKV(id int64, delta variable.TableDelta) err
 	_, err = h.ctx.(sqlexec.SQLExecutor).Execute("commit")
 	return errors.Trace(err)
 }
+
+const (
+	// StatsOwnerKey is the stats owner path that is saved to etcd.
+	StatsOwnerKey = "/tidb/stats/owner"
+	// StatsPrompt is the prompt for stats owner manager.
+	StatsPrompt = "stats"
+)
+
+func needAnalyzeTable(tbl *Table, limit time.Duration) bool {
+	if tbl.ModifyCount == 0 {
+		return false
+	}
+	t := time.Unix(0, oracle.ExtractPhysical(tbl.Version)*int64(time.Millisecond))
+	if time.Now().Sub(t) < limit {
+		return false
+	}
+	for _, col := range tbl.Columns {
+		if len(col.Buckets) > 0 {
+			return false
+		}
+	}
+	for _, idx := range tbl.Indices {
+		if len(idx.Buckets) > 0 {
+			return false
+		}
+	}
+	return true
+}
+
+// HandleAutoAnalyze analyzes the newly created table or index.
+func (h *Handle) HandleAutoAnalyze(is infoschema.InfoSchema) error {
+	dbs := is.AllSchemaNames()
+	for _, db := range dbs {
+		tbls := is.SchemaTables(model.NewCIStr(db))
+		for _, tbl := range tbls {
+			tblInfo := tbl.Meta()
+			statsTbl := h.GetTableStats(tblInfo.ID)
+			if statsTbl.Pseudo || statsTbl.Count == 0 {
```

Comment:
Also, even if the deleted table's count is not 0, we do not need to process it, because if it is deleted, then it will not be in the info schema or the stats cache.